tools/sphere.c

```diff
@@ -80,3 +78,31 @@ int do_touch( string str ) {
 int do_gaze( string str ) {
 
 }
+
+/**
+ *  Function finds and returns the sloop shadow based on the
+ *  string passed.
+ *  @param {String} sloop_type : valid values either "slooper" or "slooped"
+ *  @param {Object} ob :  player object to test for shadows
+ *  @return {Object} : returns the shadow object
+ */
+
+string find_sloop_shadow( string sloop_type, object ob ) {
+    if ( sloop_type != "slooper" || sloop_type != "slooped" ) {
+        return 0;
+    }
+    if (! ob ) {
```

Comment:
You may want to make this: `if (!objectp(ob))` just in case some rogue object decides to give you something that's not an object. Right now if that happened, you'd just go ahead and try passing it to shadow_list(), which would probably have an unintended effect (that is, it'd probably either error or return 0, which is just as bad, since you're dealing with neither possibility in the subsequent code).